test/JsonApiDotNetCoreMongoDbExampleTests/IntegrationTests/ReadWrite/Updating/Resources/UpdateResourceTests.cs

```diff
@@ -0,0 +1,416 @@
+        {
+            // Arrange
+            var existingUserAccount = _fakers.UserAccount.Generate();
+
+            await _testContext.RunOnDatabaseAsync(async db =>
+            {
+                await db.GetCollection<UserAccount>().InsertOneAsync(existingUserAccount);
+            });
+
+            var requestBody = new
+            {
+                data = new
+                {
+                    type = "userAccounts",
+                    id = existingUserAccount.StringId,
+                    attributes = new
+                    {
+                    },
+                    relationships = new
+                    {
+                    }
+                }
+            };
+
+            var route = "/userAccounts/" + existingUserAccount.StringId;
+
+            // Act
+            var (httpResponse, responseDocument) = await _testContext.ExecutePatchAsync<string>(route, requestBody);
+
+            // Assert
+            httpResponse.Should().HaveStatusCode(HttpStatusCode.NoContent);
+
+            responseDocument.Should().BeEmpty();
+        }
+
+        [Fact]
+        public async Task Can_update_resource_with_unknown_attribute()
+        {
+            // Arrange
+            var existingUserAccount = _fakers.UserAccount.Generate();
+            var newFirstName = _fakers.UserAccount.Generate().FirstName;
+
+            await _testContext.RunOnDatabaseAsync(async db =>
+            {
+                await db.GetCollection<UserAccount>().InsertOneAsync(existingUserAccount);
+            });
+
+            var requestBody = new
+            {
+                data = new
+                {
+                    type = "userAccounts",
+                    id = existingUserAccount.StringId,
+                    attributes = new
+                    {
+                        firstName = newFirstName,
+                        doesNotExist = "Ignored"
+                    }
+                }
+            };
+
+            var route = "/userAccounts/" + existingUserAccount.StringId;
+
+            // Act
+            var (httpResponse, responseDocument) = await _testContext.ExecutePatchAsync<string>(route, requestBody);
+
+            // Assert
+            httpResponse.Should().HaveStatusCode(HttpStatusCode.NoContent);
+
+            responseDocument.Should().BeEmpty();
+
+            await _testContext.RunOnDatabaseAsync(async db =>
+            {
+                var userAccountInDatabase = await db.GetCollection<UserAccount>().AsQueryable()
+                    .Where(userAccount => userAccount.Id == existingUserAccount.Id)
+                    .FirstOrDefaultAsync();
+
+                userAccountInDatabase.FirstName.Should().Be(newFirstName);
+                userAccountInDatabase.LastName.Should().Be(existingUserAccount.LastName);
+            });
+        }
+
+        [Fact]
+        public async Task Can_completely_update_resource_with_string_ID()
```

Comment:
Similar to CreateResourceTests, add a failing test when int is used for id.

Reply:
Should I add a new resource type and leave its Id as `string`? Hence, not inheriting from `MongoDbIdentifiable`?

Reply:
If you mean `int` instead of `string`, then yes.

Reply:
That's what I meant, sorry.

I wrote an initial version of the test, not putting any particular thought into what the assertions were for the time being, and I could not get past the service locator trying to instantiate an `EntityFrameworkCoreRepository`... I tried explicitly registering the `MongoDbRepository<ModelWithIntId, int>` and still I get this error...

I'll upload this in my next commit and perhaps you can help me figure out what it is I did wrong...

Reply:
I've pushed a commit to make that work.